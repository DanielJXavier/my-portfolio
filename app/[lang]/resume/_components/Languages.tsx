import { useParams } from "next/navigation";

import { getDictionary } from "get-dictionary";
import { Lang } from "i18n-config";

import { languages } from "@/[lang]/resume/_config";

import VisibilityControl from "./VisibiltyControl";

export default function Languages() {
  const { lang } = useParams<{ lang: Lang }>();

  const {
    languages: { items: languagesStrings },
    resume: { sectionTitles },
  } = getDictionary(lang);

  return (
    <section>
      <h2 className="text-[16pt] font-bold">{sectionTitles.languages}</h2>
      <hr />
      <ul className="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['â€¢'] [&_li]:before:absolute [&_li]:before:-left-[10px]">
        {languages.map(({ languageId }, i) => (
          <VisibilityControl
            key={`language-item-${i}`}
            className="items-baseline break-inside-avoid"
          >
            <div>
              <p className="text-[12pt]">
                {languagesStrings[languageId].name} (
                {languagesStrings[languageId].level})
              </p>
            </div>
          </VisibilityControl>
        ))}
      </ul>
    </section>
  );
}
