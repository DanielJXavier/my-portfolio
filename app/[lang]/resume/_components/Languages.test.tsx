import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Languages from "./Languages";

import { useParams } from "next/navigation";

jest.mock("next/navigation");

(useParams as jest.Mock).mockImplementation(() => ({ lang: "en" }));

describe("Languages component", () => {
  it("renders the component", () => {
    const { container } = render(<Languages />);

    expect(container).toMatchSnapshot();
  });
});
