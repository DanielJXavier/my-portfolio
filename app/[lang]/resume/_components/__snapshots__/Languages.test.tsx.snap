// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages component renders the component 1`] = `
<div>
  <section>
    <h2
      class="text-[16pt] font-bold"
    >
      Languages
    </h2>
    <hr />
    <ul
      class="pl-3.5 [&_li]:relative [&_li]:text-[12pt] [&_li]:text-justify [&_li]:before:content-['â€¢'] [&_li]:before:absolute [&_li]:before:-left-[10px]"
    >
      <li
        class="items-baseline break-inside-avoid flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt]"
          >
            Portuguese
             (
            Native
            )
          </p>
        </div>
      </li>
      <li
        class="items-baseline break-inside-avoid flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt]"
          >
            English
             (
            C1
            )
          </p>
        </div>
      </li>
      <li
        class="items-baseline break-inside-avoid flex gap-x-1.5 "
        data-testid="visibility-control"
        data-visible=""
      >
        <input
          checked=""
          class="print:hidden"
          type="checkbox"
        />
        <div>
          <p
            class="text-[12pt]"
          >
            Spanish
             (
            B1
            )
          </p>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
